pkg/storage/etcd3/event.go

```diff
@@ -30,14 +30,16 @@ type event struct {
 	isCreated bool
 }
 
+// parseKV converts a KeyValue and previous value to a synthetic event.
+// isCreated is true if kv.ModRevision == kv.CreateRevision OR if prevVal is nil.
 func parseKV(kv *mvccpb.KeyValue, prevVal []byte) *event {
 	return &event{
 		key:       string(kv.Key),
 		value:     kv.Value,
 		prevValue: prevVal,
 		rev:       kv.ModRevision,
 		isDeleted: false,
-		isCreated: kv.ModRevision == kv.CreateRevision,
+		isCreated: kv.ModRevision == kv.CreateRevision || prevVal == nil,
```

Comment:
since the only place we will be calling `parseKV` is from the initial sync, I think I'll just change this to not take a prevValue and always return an `isCreated:true` event, and doc as such